src/main/java/com/ibm/devops/notification/OTCAPIHelper.java

```diff
@@ -0,0 +1,249 @@
+
+            CloudFoundryClient client = new CloudFoundryClient(cloudCredentials, url, configuration, true);
+            return "bearer " + client.login().toString();
+
+        } catch (Exception e) {
+        	printStream.println("[IBM Cloud DevOps] Unexpected Exception:");
+            e.printStackTrace(printStream);
+        }
+        return "";
+    }
+    
+    /*
+     * Returns all orgs from OTC API for the given Bluemix token
+     */    
+    public static JSONObject getOrgs(EnvVars envVars, String bluemixToken, PrintStream printStream) {
+    	String orgName = envVars.get("CF_ORG");
+    	String orgsUrl = getTargetAPI(envVars) + ORGS_URL + orgName;
+
+    	CloseableHttpClient httpClient = HttpClients.createDefault();
+    	HttpGet httpGet = new HttpGet(orgsUrl);
+
+    	httpGet = addProxyInformation(httpGet);
+
+    	httpGet.setHeader("Authorization", bluemixToken);
+    	CloseableHttpResponse response = null;
+
+    	try {
+    		response = httpClient.execute(httpGet);
+    		String resStr = EntityUtils.toString(response.getEntity());
+    		if (response.getStatusLine().toString().contains("200")) {
+    			// get 200 response
+    			return JSONObject.fromObject(resStr);
+    		}
+
+    	} catch (Exception e) {
+        	printStream.println("[IBM Cloud DevOps] Unexpected Exception:");
+            e.printStackTrace(printStream);
+        }
+    	return new JSONObject();
+    }
+    
+    /*
+     * Returns all spaces from OTC API for the given Bluemix token
+     */  
+    public static JSONObject getSpaces(EnvVars envVars, String bluemixToken, PrintStream printStream) {
+    	String spaceName = envVars.get("CF_SPACE");
+    	String spacesUrl = getTargetAPI(envVars) + SPACES_URL + spaceName;
+
+    	CloseableHttpClient httpClient = HttpClients.createDefault();
+    	HttpGet httpGet = new HttpGet(spacesUrl);
+
+    	httpGet = addProxyInformation(httpGet);
+
+    	httpGet.setHeader("Authorization", bluemixToken);
+    	CloseableHttpResponse response = null;
+
+    	try {
+    		response = httpClient.execute(httpGet);
+    		String resStr = EntityUtils.toString(response.getEntity());
+    		if (response.getStatusLine().toString().contains("200")) {
+    			// get 200 response
+    			return JSONObject.fromObject(resStr);
+    		}
+
+    	} catch (Exception e) {
+        	printStream.println("[IBM Cloud DevOps] Unexpected Exception:");
+            e.printStackTrace(printStream);
+        }
+    	return new JSONObject();
+    }
+    
+    /*
+     * Returns all matching apps from OTC API for the given Bluemix token, org and space
+     */  
+    public static JSONObject getApps(EnvVars envVars, String bluemixToken, PrintStream printStream) {
+    	String orgName = envVars.get("CF_ORG");
+    	String spaceName = envVars.get("CF_SPACE");
+    	String appName = envVars.get("CF_APP");
+    	String appsUrl = getTargetAPI(envVars) + APPS_URL + appName + ORG + orgName + SPACE + spaceName;
```

Comment:
Thanks for your time reviewing this.
Fixed the typo in pom.xml (correct - no double dots :-)

Assuming I should encode all URLs using the same code as in the dra plugin:
URLEncoder.encode(<some_url>, "UTF-8").replaceAll("\\+", "%20"))